### PR TITLE
(2786) Fix `rows_for_last_financial_quarter`

### PR DESCRIPTION
**Pull request by Gweaton**
## Changes in this PR

A while back, we added protection to `rows_for_first_financial_quarter` to prevent it trying to convert `nil` values to BigDecimals when there weren't any found.

Although [the PR](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/1482) in which that was added suggests that there should always be a value for the last row of actuals, on Staging, at least, this doesn't appear to be the case.

Although the above may be due to slightly wonky, possibly invalid data on Staging, this error caused a report (with ID
b45c5b36-e502-4406-ac66-8a51321a177c) to fail being uploaded to S3 (particularly, when calling `Export::Report#rows`. We'd like to make sure this doesn't happen on Production.

## Next steps

- [ ] Run the report upload job on staging and make sure all succeeds
- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

**Comment by Gweaton** [comment]
https://trello.com/c/QKD13zKT/2786-investigate-possible-edge-case-with-report-failing-to-upload

File: spec/models/export/activity_actuals_columns_spec.rb [hunk 1]
```diff
@@ -128,6 +128,21 @@
...
+        let(:activity) { create(:project_activity) }
+        subject {
+          described_class.new(activities: [activity], include_breakdown: breakdown, report: report)
+        }
+
+        it "returns zero and not nil" do
+          last_column_data = subject.rows_for_last_financial_quarter
+          value_for_activity = last_column_data.fetch(activity.id)
+
+          expect(value_for_activity).to eq 0
```
Comment (yndajas) [inline spec/models/export/activity_actuals_columns_spec.rb:142]: Looks like there's a [be_zero matcher](https://relishapp.com/rspec/rspec-expectations/v/2-2/docs/matchers/predicate-matchers), but I don't have a particular preference for that

**Review by yndajas — APPROVED** [review]
Looks good. Do we need add any tests to the place(s) this was causing errors, or is this unit test sufficient, given it's a fairly edge case?

File: spec/models/export/activity_actuals_columns_spec.rb (hunk 1 above)
Comment (Gweaton) [reply spec/models/export/activity_actuals_columns_spec.rb:142]: I've taken this from the existing example test in the PR above, but good to know about this matcher!
